src/Compilers/Core/Portable/CryptographicHashProvider.cs

```diff
@@ -111,6 +111,43 @@ internal static HashAlgorithm TryGetAlgorithm(AssemblyHashAlgorithm algorithmId)
             }
         }
 
+        internal static IDisposable TryGetAlgorithmImpl(AssemblyHashAlgorithm algorithmId)
+        {
+            switch (algorithmId)
+            {
+                case AssemblyHashAlgorithm.None:
+                case AssemblyHashAlgorithm.Sha1:
+                    return PortableShim.SHA1.Create();
+
+                case AssemblyHashAlgorithm.Sha256:
+                    return PortableShim.SHA256.Create();
+
+                case AssemblyHashAlgorithm.Sha384:
+                    return PortableShim.SHA384.Create();
+
+                case AssemblyHashAlgorithm.Sha512:
+                    return PortableShim.SHA512.Create();
+
+                case AssemblyHashAlgorithm.MD5:
+                    return PortableShim.MD5.Create();
+
+                default:
+                    return null;
+            }
+        }
+
+        internal static object TryGetHashAlgorithmNameObj(AssemblyHashAlgorithm algorithmId)
+        {
+            switch (algorithmId)
+            {
+                case AssemblyHashAlgorithm.Sha1:
+                    return PortableShim.HashAlgorithmName.SHA1;
+
+                default:
+                    return null;
```

Comment:
Then I would say we should call this method TryGet**Sha1**HashAlgorithmNameObj and remove the parameter.